### PR TITLE
debug logs

### LINKED ISSUE #37: Basic logging
Basic of the new exactonline rate limiting policy, we were forced to review the way we do outgoing calls to exactonline, specifically we were hitting a limit on the token renewals. After checking, it turns out that your client library already manages that part really well, it just needs a different approach to store the tokens (we pretty much faced the same issue as in #30 and ended up with the same solution: a cache-backed storage).

However, before going live with this, we would like to see the change in behavior in terms of outgoing calls to exactonline. Since all of these outgoing calls are done via this library, we can't simply add log entries.

Alternatives I did think of were:
* adding in a proxy:  difficult to support he initial authorization code flow, needs to be HTTPS
* doing a full mock API: seems overkill for this purpose

Hence my question: would you be OK if we would add some log lines into the code? I would just use the standard python logging library and I would add them at `debug` level.

### PR DESCRIPTION
**Pull request by mhemeryck**
Fixes #37

Adds debug logs to track outgoing HTTP calls.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,5 @@
 /build/
 /dist/
 /.coverage
+
+tags
```
Comment (wdoekes) [inline .gitignore:8]: ?

Please leave your own tools out of my gitignore :)

File: exactonline/http.py [hunk 2]
```diff
@@ -290,6 +293,10 @@ def _http_request(url, method=None, data=None, opt=None):
     else:
         opener = request.build_opener()
 
+    logger.debug(
+        'Outgoing request for {url} '
+        'using method{method}'.format(url=url, method=method)
+    )
```
Comment (wdoekes) [inline exactonline/http.py:299]: Please use:
```
logger.debug(
    'Outgoing request for {url} using method {method}'
    .format(url=url, method=method))
```
or
```
logger.debug(
    'Outgoing request for {url} using method {method}'.format(
        url=url, method=method))
```
Needlessly breaking up a string makes it harder to grep for.

Also:
`method {method}` (with space).

**Review by wdoekes — CHANGES_REQUESTED** [review]
LGTM. Some nits in the review.

Please squash when done :)

File: .gitignore (hunk 1 above)
Comment (mhemeryck) [reply .gitignore]: np, dropped the commit

File: exactonline/http.py (hunk 2 above)
Comment (mhemeryck) [reply exactonline/http.py]: I tend to use [`black`](https://black.readthedocs.io/en/stable/) for formatting, so wasn't sure what was the best way to split. Adapted it as per your suggestion.

**Comment by mhemeryck** [comment]
Thx for the quick feedback!

**Comment by wdoekes** [comment]
:+1:

**Pull request merged**